### PR TITLE
[#1557] Migrate docs to use MarkBind V3

### LINKED ISSUE #1557: Update documentation website to use MarkBind v3
**Tell us about your environment**

* **RepoSense Version:** master
* **OS and Version:** Windows10
* **Web Browser and Version (if applicable):** Edge

**Please include the steps to reproduce the bug.**

1. Open the Dev site of RepoSense at https://reposense.org/RepoSense/

**What was expected to happen?**

The rendered layout should be similar to the layout in the production site at https://reposense.org/

**What actually happened? Please include a screenshot of the output.**

image

**If possible, include the URL to your RepoSense report or log files (if any).**

### PR DESCRIPTION
**Pull request by damithc**
Fixes #1557

Proposed commit message:
```
Migrate docs to use MarkBind V3

Our docs are not compatible with some of the breaking changes
introduced by MarkBind V3.

Let's migrate our docs to MarkBind V3 so that our website can use the
latest version of MarkBind.
```

### CODE REVIEW

**Comment by damithc** [comment]
Ready for review.

**Comment by dcshzj** [comment]
Apologies @damithc, the DevOps guide will need to be migrated as well.

**Comment by damithc** [comment]
> Apologies @damithc, the DevOps guide will need to be migrated as well.

Done. Thanks for the heads up @dcshzj

File: docs/css/main.css [hunk 1]
```diff
@@ -0,0 +1,221 @@
+mark {
+    background-color: #ff0;
+    border-radius: 5px;
+    padding-top: 0;
+    padding-bottom: 0;
+}
```
Comment (dcshzj) [inline docs/css/main.css:6]: I couldn't find any usages of this class, is this needed?

File: docs/css/main.css [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+.indented {
+    padding-left: 20px;
+}
+
+.theme-card img {
+    width: 100%;
+}
```
Comment (dcshzj) [inline docs/css/main.css:14]: I couldn't find any usages of this class, is this needed?

**Review by dcshzj — CHANGES_REQUESTED** [review]
The sidebar top heading is also hidden by the yellow notice when at small viewports, as shown in this GIF (there is supposed to be a "USER GUIDE" heading):

chrome-capture (1)

File: docs/ug/usingAuthorTags.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
-<variable name="title">Appendix: Using `@@author` tags</variable>
+{% set title = "Appendix: Using `@@author` tags" %}
 <frontmatter>
-  title: "Appendix: Using `@@author` tags"
+  title "Appendix: Using `@@author` tags"
```
[the remark this answers is not in the document]
Comment (damithc) [reply docs/ug/usingAuthorTags.md]: Yup, plus I've removed the `:` by accident. Fixed.

File: docs/ug/index.md [hunk 4]
```diff
@@ -1,11 +1,11 @@
-<variable name="title">User guide</variable>
+{% set title = "User guide" %}
 <frontmatter>
   title: "{{ title | safe }}"
 </frontmatter>
 
 {% from 'scripts/macros.njk' import embed, step, thumbnail with context %}
 
-<h1 class="display-3"><md>{{ title }}</md></h1>
+<h1 class="display-3"><md>{{ title | trim }}</md></h1>
```
[the remark this answers is not in the document]
Comment (damithc) [reply docs/ug/index.md]: Good catch @dcshzj. Removed.

File: docs/css/main.css [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+/* Bootstrap medium(md) responsive breakpoint */
+@media screen and (max-width: 991.98px) {
+    #site-nav {
+        display: none;
+    }
+
+    footer {
+        margin-left: 60px;
+    }
```
[the remark this answers is not in the document]
Comment (damithc) [reply docs/css/main.css:116]: This part of the css is generated by the `makbind init` command. I've raised this on their side.

File: docs/css/main.css [hunk 6]
```diff
@@ -0,0 +1,221 @@
+mark {
+    background-color: #ff0;
+    border-radius: 5px;
+    padding-top: 0;
+    padding-bottom: 0;
+}
+
+.indented {
+    padding-left: 20px;
+}
```
[the remark this answers is not in the document]
Comment (damithc) [reply docs/css/main.css:10]: Removed.

File: docs/css/main.css (hunk 1 above)
Comment (damithc) [reply docs/css/main.css:6]: I think this class is used for highlighting (i.e., `==word==` syntax). Anyway, this part of the css is generated by the `makbind init` command.

File: docs/css/main.css (hunk 2 above)
Comment (damithc) [reply docs/css/main.css:14]: This part of the css is generated by the `makbind init` command. Best to leave it be.

**Comment by damithc** [comment]
Thanks for the review @dcshzj I've updated as per your comments.

**Comment by damithc** [comment]
> The sidebar top heading is also hidden by the yellow notice when at small viewports, as shown in this GIF (there is supposed to be a "USER GUIDE" heading):

I've raised this on MarkBind side.

**Comment by dcshzj** [comment]
> I've raised this on MarkBind side.

Sorry @damithc, is this on MarkBind's issue tracker? I could not find a relevant issue posted.

**Comment by damithc** [comment]
> > I've raised this on MarkBind side.
> 
> Sorry @damithc, is this on MarkBind's issue tracker? I could not find a relevant issue posted.

It's in their discussions page https://github.com/MarkBind/markbind/discussions/1631#discussioncomment-986162

**Comment by damithc** [comment]
@reposense/active-devs , remember to merge this PR. Go ahead and do further changes as necessary.

**Review by dcshzj — APPROVED** [review]
LGTM! It is unlikely that the two issues raised will be resolved anytime soon, but it does not affect our documentation too significantly.

**Review by fzdy1914 — APPROVED** [review]

**Pull request merged**